lib/Onyx.js

```diff
@@ -455,6 +492,52 @@ function merge(key, val) {
         });
 }
 
+/**
+ * Merges a collection based on their keys
+ *
+ * @param {String} collectionKey
+ * @param {Object} collection
+ * @returns {Promise}
+ */
+function mergeCollection(collectionKey, collection) {
+    // Confirm all the collection keys belong to the same parent
+    _.each(collection, (data, dataKey) => {
+        if (!isKeyMatch(collectionKey, dataKey)) {
+            throw new Error(`Provided collection does not have all its data belonging to the same parent. CollectionKey: ${collectionKey}, DataKey: ${dataKey}`);
+        }
+    });
+
+    const existingKeyCollection = {};
+    const newCollection = {};
+    return AsyncStorage.getAllKeys()
+        .then((keys) => {
+            _.each(collection, (data, dataKey) => {
+                if (keys.includes(dataKey)) {
+                    existingKeyCollection[dataKey] = data;
+                } else {
+                    newCollection[dataKey] = data;
+                }
+            });
+
+            const keyValuePairsForExistingCollection = _.reduce(existingKeyCollection, (finalArray, val, key) => ([
+                ...finalArray,
+                [key, JSON.stringify(val)],
+            ]), []);
+            const keyValuePairsForNewCollection = _.reduce(newCollection, (finalArray, val, key) => ([
+                ...finalArray,
+                [key, JSON.stringify(val)],
+            ]), []);
+
+            // New keys will be added via multiSet while existing keys will be updated using multiMerge
```

Comment:
Maybe this would be a good spot to add some "why" context? Probably there is no harm in adding 

>because setting a key that doesn't exist yet with `multiMerge` will throw errors